libs/wingc/src/type_check.rs

```diff
@@ -773,6 +794,24 @@ impl TypeRef {
 		}
 	}
 
+	pub fn is_resource(&self) -> bool {
+		if let Type::Resource(_) = **self {
+			true
+		} else {
+			false
+		}
```

Comment:
Perhaps change to this and also all the other "is_XXX" methods here.
```suggestion
		matches!(**self, Type::Resource(_))
```